### ISSUE TITLE
Issue with New Api Hadoop file 

### ISSUE THREAD

**Issue by bischofs** [valtech/ros_hadoop#14]
Having an issue loading the jar for RosbagInputFormat following your instructions, maybe it is from the classpath loading? Do every one of the jars need to be in the hadoop classpath as well as each spark executor? 

![screenshot from 2018-08-13 09-52-51](https://user-images.githubusercontent.com/1606648/44036077-f4c0a40c-9ede-11e8-882a-083f227dac2f.png)

**Comment by wiegelmann** [comment]
The jar does not need to be in the Hadoop classpath, but the Spark driver will take care of it if configured correctly.

This exception looks interesting indeed. Can you send some more information about the cluster configuration and Scala / Spark versions used?

The provided precompiled jar contains the Scala version in it's name, if the user uses a different Scala/Spark version we highly recommend to rebuild it using the SBT build script provided, after changing the Scala version accordingly. Very old Scala version are not supported.

**Comment by bischofs** [comment]
Yeah I added the jar to the hadoop classpath and then I got that exception.

Cluster is a standard dataproc from google cloud, I tried searching for that error and most of the references are toward classpath issues with spark. 

![spark](https://user-images.githubusercontent.com/1606648/44055301-01d0a280-9f13-11e8-9703-13920d4cbd8f.png)

**Comment by bischofs** [comment]
Okay I figured most of it out, I tried it with the pyspark --jars and included all the jars and it worked, to get it working with jupyter I just tacked on the --jars in the kernel.json for the pyspark kernel. So its for sure a jar loading issue even if its quite cryptic.  

On another note spark seems to complain if you don't put the index file where all the executors can access it, so I used sc.addFile() and that seems to work, tho I'm having issues because apparently the executors doesn't have access to a python binary. Will keep tinkering with it.
